Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailController.cs

```diff
@@ -0,0 +1,197 @@
+        {
+            if (!inputData.UserOwnedReels) return;
+
+            inputData.ExecuteDeleteAction(currentReelIndex);
+            isClosing = true;
+        }
+
+        protected override void OnViewInstantiated()
+        {
+            metadataSidePanelAnimator = new MetadataSidePanelAnimator(viewInstance!.rootContainer, viewInstance.infoButtonImageRectTransform);
+        }
+
+        protected override void OnViewClose()
+        {
+            viewInstance!.infoButton.onClick.RemoveListener(ToggleInfoSidePanel);
+            viewInstance!.previousScreenshotButton.onClick.RemoveListener(ShowPreviousReel);
+            viewInstance!.nextScreenshotButton.onClick.RemoveListener(ShowNextReel);
+            viewInstance!.downloadButton.onClick.RemoveListener(DownloadReelClicked);
+            viewInstance!.linkButton.onClick.RemoveListener(CopyReelLinkClicked);
+            viewInstance!.twitterButton.onClick.RemoveListener(ShareReelClicked);
+            viewInstance!.deleteButton.onClick.RemoveListener(ShowDeleteModal);
+            HideDeleteModal();
+
+            viewInstance.mainImageCanvasGroup.alpha = 0;
+            photoDetailInfoController.Release();
+        }
+
+        protected override void OnBeforeViewShow()
+        {
+            currentReelIndex = inputData.CurrentReelIndex;
+            viewInstance!.deleteButton.gameObject.SetActive(inputData.UserOwnedReels);
+            viewInstance!.previousScreenshotButton.gameObject.SetActive(false);
+            viewInstance!.nextScreenshotButton.gameObject.SetActive(false);
+            isClosing = false;
+        }
+
+        private void DownloadReelClicked() =>
+            ReelCommonActions.DownloadReel(inputData.AllReels[currentReelIndex].url, webBrowser);
+
+        private void CopyReelLinkClicked() =>
+            ReelCommonActions.CopyReelLink(inputData.AllReels[currentReelIndex].id, decentralandUrlsSource, systemClipboard);
+
+        private void ShareReelClicked() =>
+            ReelCommonActions.ShareReelToX(shareToXMessage, inputData.AllReels[currentReelIndex].id, decentralandUrlsSource, systemClipboard, webBrowser);
+
+        private void ShowPreviousReel()
+        {
+            currentReelIndex = Math.Clamp(currentReelIndex - 1, 0, inputData.AllReels.Count - 1);
+            ShowReel(currentReelIndex);
+        }
+
+        private void ShowNextReel()
+        {
+            currentReelIndex = Math.Clamp(currentReelIndex + 1, 0, inputData.AllReels.Count - 1);
+            ShowReel(currentReelIndex);
+        }
+
+        private void ShowReel(int reelIndex)
+        {
+            CheckNavigationButtonVisibility(inputData.AllReels, reelIndex);
+            showReelCts = showReelCts.SafeRestart();
+            ShowReelAsync(reelIndex, showReelCts.Token).Forget();
+        }
+
+        private async UniTaskVoid ShowReelAsync(int reelIndex, CancellationToken ct)
+        {
+            viewInstance!.mainImageCanvasGroup.alpha = 0;
```

Comment:
Same as previous comments. The view should provide a method that enables the loading spinner, for example.